ui/src/components/login/OktaSignInWidget.tsx

```diff
@@ -0,0 +1,36 @@
+import { useEffect } from 'react';
+import OktaSignIn from '@okta/okta-signin-widget';
+
+interface OktaSignInWidgetProps {
+  onSuccess: (tokens: any) => void;
+  onError: (error: Error) => void;
+}
+
+export default function OktaSignInWidget({
+  onSuccess,
+  onError,
+}: OktaSignInWidgetProps) {
+  useEffect(() => {
+    const widget = new OktaSignIn({
+      clientId: process.env.REACT_APP_OKTA_CLIENT_ID,
+      redirectUri: window.location.origin,
+      issuer:
+        'https://' + process.env.REACT_APP_OKTA_DOMAIN + '/oauth2/default',
+      logo: '/uc-logo.png',
+    });
+    widget
+      .showSignInToGetTokens({
+        el: '#osw-container',
+      })
+      .then(function (res) {
+        onSuccess(res?.idToken?.idToken);
+        widget.remove();
+      })
+      .catch(function (error) {
+        onError(error);
+      });
+    return () => widget.remove();
+  }, [onSuccess, onError]);
```

Comment:
I also added onSuccess and onError to the dependency array in my Gist, so I share some responsibility. However, since we want this hook to run only once on mount, can we make the dependency array empty? With the current implementation, it ends up making multiple API calls.

```suggestion
  // eslint-disable-next-line react-hooks/exhaustive-deps
  }, []);
```